src/NuGet.Core/NuGet.Build.Tasks.Pack/Strings.resx

```diff
@@ -169,6 +169,10 @@
   <data name="IsPackableFalseError" xml:space="preserve">
     <value>This project cannot be packaged because packaging has been disabled. Add &lt;IsPackable&gt;true&lt;/IsPackable&gt; to the project file to enable producing a package from this project.</value>
   </data>
+  <data name="MissingRequiredDot" xml:space="preserve">
+    <value>Dots in TargetFramework versions are required. The missing dot in '{0}' might cause future breakage.</value>
```

Comment:
@JonDouglas do you have comments about this user-facing error message?

Reply:
@zkat & @zivkan 

`Dots in TargetFramework versions are required. The missing dot in '{0}' might cause future breakage.`

->

`The TargetFrameworkVersion value `{0}` was not recognized. It may be missing a dotted version number.

`The TargetFrameworkVersion value `{0}` was not recognized. It may be missing a version number separated by periods.

or

`The TargetFrameworkVersion value `{0}` is not valid. Ensure your TargetFrameworkVersion includes a dotted version number.

`The TargetFrameworkVersion value `{0}` is not valid. Ensure your TargetFrameworkVersion includes a version number separated by periods.

Reply:
The framework is technically valid and understood...just might not work the way that customer intended it to. 

In particular the problem is that `net50` is parses to version `5.0`, and the eventual progression of NET will eventually lead to version 10.0, so if people still write frameworks with versions then `net10`, it'd get parsed to `1.0` while they might expect it to parse to `10.0`

Reply:
Yep, and we want to change the behavior of the customer to promote the best practice.

This is similar with mobile TFMs such as monoandroid10 vs. monoandroid10.0. One is 1.0 & the other is 10.0. 

So TargetFramework = monoandroid, TargetFrameworkVersion = v1.0

If anything, we should be promoting the separator across the board, it's the best practice is it not? Everything in the NET 5 spec suggests to use a period separator for all `TargetBlahVersion` properties.

https://github.com/dotnet/designs/blob/master/accepted/2020/net5/net5.md#upgrading-the-os-bindings

https://github.com/dotnet/designs/blob/master/accepted/2020/net5/net5.md#lighting-up-on-later-os-versions

The bigger question is best practices for the `<TargetFramework>` element here. Is it period separator or not? Prior to .NET Core it looks like no, it doesn't matter. After .NET Core it seems that it should be included. Thoughts?

Reply:
Separator across the board is the right approach. 

> Everything in the NET 5 spec suggests to use a period separator for all TargetBlahVersion properties.

Those are not user defined usually. 
The idea is that the TargetFramework property can be *anything*. 
NuGet doesn't parse it, which is why the check is only happening at pack time.

Reply:
Shouldn't we create an issue on the SDK team, or contribute a pull request ourselves, to do it in the SDK instead? It doesn't make much sense to me to do at pack time. It should be done at restore, but only when the monikers were not provided and the TargetFramework property was parsed, hence why it can only be done in the SDK and not in NuGet.

Reply:
@zivkan So what I'm hearing is that the suggestion is to drop this PR in favor of doing this SDK-side in a more general way?

As far as whether to warn always or warn only for certain frameworks -- I think the concern was that this could get very noisy, very quickly for people. I mean even for us, we'd need to change a bunch of `net45`/`net472` items in our tests and projects?

Reply:
yes, my opinion is close this PR, and someone should make a change in the SDK instead.

The key is, do not warn if my csproj has:
```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFrameworks>whatever</TargetFrameworks>
  </PropertyGroup>

  <PropertyGroup Condition=" '$(TargetFramework)' == 'whatever' ">
    <TargetFrameworkIdentifier>.NETCoreApp</TargetFrameworkIdentifier>
    <TargetFrameworkVersion>v5.0</TargetFrameworkIdentifier>
    <TargetFrameworkMoniker>$(TargetFrameworkIdentifier),Version=$(TargetFrameworkVersion)</TargetFrameworkMoniker>
  </PropertyGroup>
</Project>
```

This csproj doesn't have a `.` in the TargetFramework value, but it doesn't need one because it correctly sets the 3 other properties needed to correctly define the canonical target framework. Only when the SDK parses the TargetFramework value to set the other properties itself, in that situation warn about missing dots.

And yes, you're right about the noisy warnings. Just like this PR, we should only warn when the TFM ends up being TFI='.NETCoreApp' and TFV >= '5.0'. Or any TFI='.NETStandard' apparently.

Reply:
I think we should still do it on our side in pack because people can manually create packages. We should catch things like that in the nuspec and package folders.